network-api/networkapi/wagtailpages/factory/bannered_campaign_page.py

```diff
@@ -0,0 +1,54 @@
+from factory import (
+    Trait,
+    SubFactory
+)
+from wagtail.core.models import Page as WagtailPage
+
+from networkapi.wagtailpages.models import BanneredCampaignPage
+from networkapi.utility.faker.helpers import (
+    reseed,
+    get_homepage
+)
+from .petition import PetitionFactory
+from .mini_site_namespace import MiniSiteNamespaceFactory
+from .donation import DonationModalsFactory
+from .abstract import CMSPageFactory
+from .campaign_page import CampaignPageFactory
+
+
+class BanneredCampaignPageFactory(CMSPageFactory):
+    class Meta:
+        model = BanneredCampaignPage
+
+    class Params:
+        no_cta = Trait(cta=None)
+
+    cta = SubFactory(PetitionFactory)
+
+
+def generate(seed):
+    home_page = get_homepage()
+    reseed(seed)
+
+    try:
+        bannered_campaign_namespace = WagtailPage.objects.get(title='bannered_campaigns')
+        print('bannered campaigns namespace exists')
+    except WagtailPage.DoesNotExist:
+        print('Generating a bannered campaigns namespace')
+        bannered_campaign_namespace = MiniSiteNamespaceFactory.create(
+            parent=home_page,
+            title='bannered_campaigns',
+            live=False
+        )
+
+    reseed(seed)
+
+    print('Generating Bannered Campaign Pages under namespace')
+    campaigns = [CampaignPageFactory.create(parent=bannered_campaign_namespace) for i in range(5)]
```

Comment:
```suggestion
        BanneredCampaignPageFactory.getOrCreate(
          parent=bannered_campaign_namespace
          title="Test Bannered Campaign"
        )
```

Reply:
because we want to build a banneredcampaign, not a standard campaign. Also we only need 1, rather than 5